api/src/main/java/bio/terra/tanagra/app/controller/EntityCountsApiController.java

```diff
@@ -0,0 +1,57 @@
+package bio.terra.tanagra.app.controller;
+
+import bio.terra.tanagra.generated.controller.EntityCountsApi;
+import bio.terra.tanagra.generated.model.ApiGenerateCountsSqlQueryRequest;
+import bio.terra.tanagra.generated.model.ApiSearchEntityCountsRequest;
+import bio.terra.tanagra.generated.model.ApiSearchEntityCountsResponse;
+import bio.terra.tanagra.generated.model.ApiSqlQuery;
+import bio.terra.tanagra.service.databaseaccess.QueryResult;
+import bio.terra.tanagra.service.query.EntityCounts;
+import bio.terra.tanagra.service.query.QueryService;
+import bio.terra.tanagra.service.query.api.ApiConversionService;
+import bio.terra.tanagra.service.query.api.QueryResultConverter;
+import javax.validation.Valid;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+
+/** An {@link EntityCountsApi} controller for operating on entity counts. */
+@Controller
+public class EntityCountsApiController implements EntityCountsApi {
+  private final ApiConversionService apiConversionService;
+  private final QueryService queryService;
+
+  @Autowired
+  public EntityCountsApiController(
+      ApiConversionService apiConversionService, QueryService queryService) {
+    this.apiConversionService = apiConversionService;
+    this.queryService = queryService;
+  }
+
+  @Override
+  public ResponseEntity<ApiSqlQuery> generateCountsSqlQuery(
+      String underlayName, String entityName, @Valid ApiGenerateCountsSqlQueryRequest body) {
+    // TODO authorization check.
+    EntityCounts entityCounts =
+        apiConversionService.convertEntityCounts(underlayName, entityName, body.getEntityCounts());
+    String sql = queryService.generateSql(entityCounts);
+    return ResponseEntity.ok(new ApiSqlQuery().query(sql));
+  }
+
+  @Override
+  public ResponseEntity<ApiSearchEntityCountsResponse> searchEntityCounts(
+      String underlayName,
+      String entityName,
+      ApiSearchEntityCountsRequest body,
+      Integer pageSize,
```

Comment:
pagination parameters appear to be ignored; I would probably omit these from the API spec if they are not supported yet

Reply:
Removed. I had included these to be consistent with the existing entityInstances API endpoints, but agreed it's probably better to just omit them until we're ready to support them.